traitsui/wx/file_editor.py

```diff
@@ -122,7 +122,7 @@ def dispose(self):
         """ Disposes of the contents of an editor.
         """
         panel = self.control
-        panel.Bind(wx.EVT_BUTTON, self._button.GetId(), None)
+        panel.Bind(wx.EVT_BUTTON, None, self._button.GetId())
```

Comment:
Since this is using the id of the button rather than the instance, you need to pass the id in via keyword argument.
```suggestion
        panel.Bind(wx.EVT_BUTTON, None, id=self._button.GetId())
```